tabular/src/autogluon/tabular/register/_model_register.py

```diff
@@ -0,0 +1,157 @@
+        if model_cls.ag_key is None:
+            raise AssertionError(
+                f"Cannot add model_cls with `ag_key=None`. "
+                f"Ensure you set class attribute `ag_key` to a string for your model_cls: {model_cls}"
+                f'\n\tFor example, LightGBModel sets `ag_key = "GBM"`'
+            )
+        if model_cls.ag_name is None:
+            raise AssertionError(
+                f"Cannot add model_cls with `ag_name=None`. "
+                f"Ensure you set class attribute `ag_name` to a string for your model_cls: {model_cls}"
+                f'\n\tFor example, LightGBModel sets `ag_name = "LightGBM"`'
+            )
+        assert isinstance(model_cls.ag_key, str)
+        assert isinstance(model_cls.ag_name, str)
+        assert isinstance(model_cls.ag_priority, int)
+        if model_cls.ag_key in self._key_to_cls_map:
+            raise AssertionError(
+                f"Cannot register a model class that shares a model key with an already registered model class."
+                f"\n`model_cls.ag_key` must be unique among registered models:"
+                f"\n\t        New  Class: {model_cls}"
+                f"\n\tConflicting  Class: {self._key_to_cls_map[model_cls.ag_key]}"
+                f"\n\tConflicting ag_key: {model_cls.ag_key}"
+            )
+        self._model_cls_list.append(model_cls)
+        self._key_to_cls_map[model_cls.ag_key] = model_cls
+
+    def remove(self, model_cls: Type[AbstractModel]):
+        """
+        Removes `model_cls` from the model register
+        """
+        assert self.exists(model_cls), f"Cannot remove model_cls that isn't registered: {model_cls}"
+        self._model_cls_list = [m for m in self._model_cls_list if m != model_cls]
+        self._key_to_cls_map.pop(model_cls.ag_key)
+
+    @property
+    def model_cls_list(self) -> list[Type[AbstractModel]]:
+        return self._model_cls_list
+
+    @property
+    def keys(self) -> list[str]:
+        return [self.key(model_cls) for model_cls in self.model_cls_list]
+
+    def key_to_cls_map(self) -> dict[str, Type[AbstractModel]]:
+        return self._key_to_cls_map
+
+    def key_to_cls(self, key: str) -> Type[AbstractModel]:
+        if key not in self._key_to_cls_map:
+            raise ValueError(
+                f"No registered model exists with provided key: {key}"
+                f"\n\tValid keys: {list(self.key_to_cls_map().keys())}"
+            )
+        return self.key_to_cls_map()[key]
+
+    def priority_map(self, problem_type: str | None = None) -> dict[Type[AbstractModel], int]:
+        return {model_cls: self.priority(model_cls, problem_type=problem_type) for model_cls in self._model_cls_list}
+
+    def key_to_priority_map(self, problem_type: str | None = None) -> dict[str, int]:
+        return {self.key(model_cls): self.priority(model_cls, problem_type=problem_type) for model_cls in self._model_cls_list}
```

Comment:
It is convenient to have both the `alias -> *` and `cls -> *` mappings available here, but I think that for maintainability, it might actually be better to only keep one (either class or alias). This way future users / contributors won't need to guess which of these needs to be used, and there will be less code to be maintained.

Reply:
fair, I've removed the key_to_priority and key_to_priority_map functions